### ISSUE TITLE
Add Simpler control to change Crossfader Contour

### ISSUE THREAD

**Issue by Swiftb0y** [mixxxdj/mixxx#13363]
### Feature Description

Most controllers only have a single knob for switching between a constant-power and a "scratch" curve. The current split approach (one scratch mode and a separate constant power mode) makes it awkward to map these single knobs. 

For reference, here's a couple screenshots from existing hardware only featuring a single knob (somtimes even just a switch):
Numark NS6II:
![image](https://github.com/mixxxdj/mixxx/assets/12380386/d853fd21-74ce-4546-bc94-75df9a25c9ef)
Pioneer DJM
![image](https://github.com/mixxxdj/mixxx/assets/12380386/fa51a7ba-b676-477e-a2ed-c30484f26f77)
Denon X1850
![image](https://github.com/mixxxdj/mixxx/assets/12380386/f078fc7b-675b-499b-9d87-ec641ea3b6da)
Traktor Kontrol S4 MK3:
![image](https://github.com/mixxxdj/mixxx/assets/12380386/ac652ab0-e089-4a48-a72a-b1648a5cdd11)

**Comment by ronso0** [comment]
FWIW there is already a helper in common-controller-scripts.js used by a few controllers:
https://github.com/mixxxdj/mixxx/blob/3d8e7fec8ba03fac34769bb2de6e863fd67b3efa/res/controllers/common-controller-scripts.js#L428

Are you suggesting that this new control should manipulate all three `[Mixer Profile]` controls `xFaderMode`, `xFaderCalibration` and `xFaderCurve`?

**Comment by Swiftb0y** [comment]
well, `script.crossFaderCurve` does the "one-knob" translation, but it produces a different curve than what is expected by these device (and usually also modeled in the hardware mixer). So yeah, it would need something that manipulates all three smoothly (or probably some other function entirely).

**Comment by JoergAtGithub** [comment]
Just for reference: The most complex crossfader control might be the one of the Vestax PMC-05Pro D https://www.vestax.net/file/ve-pmc-05pro-d-manual-pdf But I guess nobody has a real need for this functionality.

**Comment by ronso0** [comment]
> But I guess nobody has a real need for this functionality.

Looks fun 😆 
![Image](https://github.com/user-attachments/assets/31251f39-f8b9-4cea-a9fc-d315dcc5887a)

**Comment by ronso0** [comment]
I dug a bit deeper, I'll try to summarize what's obvious -- and what's not:

In  the **mixer preferences** we have only one slider (and a mode switch Addit. / ConstPwr)
The slider [sets the `transform` value](https://github.com/mixxxdj/mixxx/blob/91100d74034a278eb02d790852bc0267d6011ec7/src/preferences/dialog/dlgprefmixer.cpp#L911) (= `xFaderCurve` CO) -- `calibration` (= `xFaderCalibration` CO) is then calculated by `EngineXfader::getPowerCalibration(transform)`
> // get the transform_root of -3db (.5)
    return pow(0.5, 1.0 / transform);
which always gives us smooth xfade curve.

Okay, so with the GUI there is no way to adjust the `xFaderCalibration` CO.
And it's really [needed only for CONST_PWR mode](https://github.com/mixxxdj/mixxx/blob/91100d74034a278eb02d790852bc0267d6011ec7/src/engine/enginexfader.cpp#L31)

So why would mappings want to set it to anything else than `pow(0.5, 1.0 / xFaderCurve)`?
For the fun of it I added a 'Calibration' slider and use its value to render the curves (and, for comparison, print what `getPowerCalibration()` returns for the current "curve" value)
Turns out you can easily produce ... adventurous xfade curves like these 😉 
![Image](https://github.com/user-attachments/assets/e0e5acf3-154c-4b5a-9447-3b5008656739) ![Image](https://github.com/user-attachments/assets/1ea182f8-1c80-4f7a-893a-943740c14ad2)

And this. Which, while it looks somehow reasonable, kind of defeats the purpose of the 'Mixing / Scratching' curve, no?
![Image](https://github.com/user-attachments/assets/eb244afb-0a31-4475-b669-f7686fce0022)

This makes me believe the `[MixerProfile],xFaderCalibration` CO is used the "wrong way" in many mappings.
[A few built-in mappings](https://github.com/search?q=repo%3Amixxxdj%2Fmixxx+%22xFaderCalibration%22+path%3A%2F%5Eres%5C%2Fcontrollers%5C%2F%2F&type=code) use it, too
<sub>(exceptions are the Numark N4 and the NS6II (re-implements EngineXfader::getPowerCalibration()). But others either use magic numbers or do weird (intransparent) scaling, like the Reloop Jockey 3 ME which also sets `xFaderCalibration` with caring about `xFaderCurve`.</sub>

What makes all this even more fun:  **Mixer preferences**
* the xfader graphic  show's the curve based on **config** values -- not COs (possible fix is #14124 )
* and calculates `calibration` on its own
* and overwrites the mapping's curve anyway when Apply is pressed 

Besides, I'm not sure why [`script.crossfaderCurve` uses other bounds](https://github.com/mixxxdj/mixxx/blob/3d8e7fec8ba03fac34769bb2de6e863fd67b3efa/res/controllers/common-controller-scripts.js#L433C1-L433C23) than the xfader.. ControlPotmeters?
And why is [`xFaderCalibration` allowed to go out-of-bounds?](https://github.com/mixxxdj/mixxx/blob/e9cf7748458750f470f8988df4553b28fa59263f/src/engine/enginemixer.cpp#L116)

**Comment by ronso0** [comment]
That said, I'll check if I can easily ™ integrate xFadercalibration into #14124, as well.

**Comment by Swiftb0y** [comment]
Yeah either way, we there definitely is reason for a simpler control to exist, do you agree?

**Comment by Swiftb0y** [comment]
> FWIW with https://github.com/mixxxdj/mixxx/pull/14124 it'll be much easier for mapping developers to find the parameters for the desired curve.
WYSIWYG

Yeah, much better. I'll take a look at that PR again asap.

Don't really have an opion on 2. and 3.

**Comment by JoergAtGithub** [comment]
Can we use another term than "Calibration", as many controllers and mixers have calibration feature to compensate the failure of the fader hardware (usually the end and center position values of a fader are corrected by this calibration).

**Comment by ronso0** [comment]
The CO is `xFaderCalibration` so 🤷 
Right now it's neither documented nor appears in the preferences GUI (and there are no plans to expose it there, and even then I'd say the risk of confusion is minimal because it's in the context 'Crossfader Curve', with or without controller attached)
